overlord/snapstate/snapstate.go

```diff
@@ -2183,6 +2061,52 @@ func autoRefreshPhase2(ctx context.Context, st *state.State, updates []*refreshC
 	return tasksets, nil
 }
 
+// checkDiskSpace checks if there is enough space for the requested snaps and their prerequisites
+func checkDiskSpace(st *state.State, changeKind string, infos []minimalInstallInfo, userID int) error {
+	var featFlag features.SnapdFeature
+
+	switch changeKind {
+	case "install":
+		featFlag = features.CheckDiskSpaceInstall
+	case "refresh":
+		featFlag = features.CheckDiskSpaceRefresh
+	default:
+		panic(fmt.Sprintf("unsupported disk space check for change of kind %q", changeKind))
```

Comment:
I would prefer an internal error here instead of a panic mostly to ensure the daemon stays alive in cases like this.